x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/rest/RestRepositoryStatsAction.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.searchablesnapshots.rest;
+
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.RestHandler;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.action.RestToXContentListener;
+import org.elasticsearch.xpack.searchablesnapshots.action.RepositoryStatsAction;
+import org.elasticsearch.xpack.searchablesnapshots.action.RepositoryStatsRequest;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.elasticsearch.rest.RestRequest.Method.GET;
+
+public class RestRepositoryStatsAction extends BaseRestHandler {
```

Comment:
Just a general question (will try to review properly shortly):  Should this maybe live in x-pack core since this isn't just a searchable snapshots related piece of functionality and we have plans to use it elsewhere also?

Reply:
I wanted to keep the scope small here so that we can iterate more quickly on this API. For now, the scope of this is only targeting searchable snapshot investigations. We can at a later stage discuss whether this change is more broadly usable. The main issue with the stats reported here are that they're reset as soon as a repository is updated or a node is restarted, which might make this a confusing API. The core functionality for tracing, that might be useful in the context of Cloud, is in the respective blobstore implementation, and might be exposed differently (e.g. periodic logging)